app/javascript/src/components/Dashboard/Settings/Profile.jsx

```diff
@@ -9,25 +9,37 @@ import profilesApi from "apis/profiles";
 import { useUserState } from "contexts/user";
 import { useUserDispatch } from "contexts/user";
 
+import ConfirmPasswordModal from "./ConfirmPasswordModal";
 import { PROFILE_FORM_VALIDATION_SCHEMA } from "./constants";
 import { buildProfileFormInitialValues } from "./utils";
 
 const Profile = () => {
+  const [showPasswordModal, setShowPasswordModal] = useState(false);
   const [submitted, setSubmitted] = useState(false);
   const { user } = useUserState();
   const userDispatch = useUserDispatch();
 
-  const handleSubmit = async data => {
+  const handleSubmit = async (data, { resetForm }) => {
     try {
       const {
         data: { user },
       } = await profilesApi.updateProfile(data);
       userDispatch({ type: "SET_USER", payload: { user } });
     } catch (err) {
       logger.error(err);
+    } finally {
+      resetForm("password");
```

Comment:
If we are resetting form over, then I think we can add enableReinitialize to the formik form.

- https://formik.org/docs/api/formik#enablereinitialize-boolean
- https://stackoverflow.com/questions/58775482/how-to-reset-form-in-formik-using-enablereinitialize

Reply:
@yedhink Once the form is submitted the response could be one of the following:
- Validation fails for first name or last name. In that case, we don't need to reinitialize.
- Everything goes well and the update is successful. I believe this is the actual use case of `enableReinitialize` since the values have been updated. In this case, we do not need to reinitialize the form because the form will reflect updated values. If user refreshes the page, form will be reinitialized with the latest values.
- Password is incorrect. In that case, we should only reset the password.

Please let me know if I am missing something here.

Reply:
@rishabhanand12 https://www.loom.com/share/330e7fb58bc94a4984cc28b89f9e266b

Reply:
@yedhink Please take a look.

Reply:
@rishabhanand12 
<img width="334" alt="image" src="https://user-images.githubusercontent.com/31244999/153219259-0496de31-0f66-4092-8ff1-2313541589f1.png">

Not sure what you meant.

Reply:
Sorry I forgot to add the link. Here it is https://www.loom.com/share/441a92408315477da67823afe644f37c.